text_extensions_for_pandas/array/arrow_conversion.py

```diff
@@ -133,10 +132,15 @@ def span_to_arrow(char_span: SpanArray) -> pa.ExtensionArray:
     begins_array = pa.array(char_span.begin)
     ends_array = pa.array(char_span.end)
 
-    typ = ArrowSpanType(begins_array.type, char_span.target_text)
+    # Create a dictionary array from StringTable used in this span
+    dictionary = pa.array(list(char_span._string_table.things))
+    target_text_dict_array = pa.DictionaryArray.from_arrays(char_span._text_ids, dictionary)
+    # TODO: remove unused things and normalize text_ids?
```

Comment:
Not sure if we should sanitize these before serializing to remove any unused document, then reset the text ids?